packages/react/src/primitives/Autocomplete/useAutocomplete.tsx

```diff
@@ -0,0 +1,251 @@
+    React.useCallback(
+      (event) => {
+        const { key } = event;
+
+        if (!DEFAULT_KEYS.has(key)) {
+          return;
+        }
+
+        event.preventDefault();
+
+        switch (key) {
+          case ESCAPE_KEY:
+            if (isMenuOpen) {
+              setIsMenuOpen(false);
+              setActiveIdx(-1);
+            } else {
+              handleOnClear();
+            }
+            break;
+          case ENTER_KEY:
+            if (!activeOption) {
+              if (isFunction(onSubmit)) {
+                onSubmit(composedValue);
+              }
+            } else {
+              const { label } = activeOption;
+              if (!isControlled) {
+                setInternalValue(label);
+              }
+
+              if (isFunction(onSelect)) {
+                onSelect(activeOption);
+              }
+            }
+            setIsMenuOpen(false);
+            setActiveIdx(-1);
+            break;
+          case ARROW_DOWN:
+            if (filteredOptions.length <= 0) {
+              return;
+            }
+            setIsMenuOpen(true);
+            setActiveIdx(
+              activeIdx >= filteredOptions.length - 1 ? 0 : activeIdx + 1
+            );
+            break;
+          case ARROW_UP:
+            if (filteredOptions.length <= 0) {
+              return;
+            }
+            setIsMenuOpen(true);
+            setActiveIdx(
+              activeIdx <= 0 ? filteredOptions.length - 1 : activeIdx - 1
+            );
+        }
+      },
+      [
+        activeIdx,
+        activeOption,
+        composedValue,
+        filteredOptions,
+        handleOnClear,
+        isControlled,
+        isMenuOpen,
+        onSelect,
+        onSubmit,
+      ]
+    );
+
+  React.useEffect(() => {
+    const menuElement = document.getElementById(menuId);
+    if (menuElement && isMenuOpen) {
+      const { top, bottom } = menuElement.getBoundingClientRect();
+
+      if (top < 0 || bottom > document.documentElement.clientHeight) {
+        window.scrollTo({
+          top:
+            bottom -
+            document.documentElement.clientHeight +
+            window.scrollY +
+            20,
+          behavior: 'smooth',
+        });
+      }
+    }
+  }, [isMenuOpen, menuId]);
+
+  React.useEffect(() => {
+    const listboxElement = document.getElementById(listboxId);
+    const activeOptionElement = document.getElementById(activeOptionId);
+
+    if (activeOptionElement && listboxElement) {
+      const { scrollTop, clientHeight } = listboxElement;
+      const { offsetHeight, offsetTop } = activeOptionElement;
+      const { top, bottom } = activeOptionElement.getBoundingClientRect();
+
+      if (scrollTop > offsetTop) {
+        listboxElement.scrollTop = offsetTop;
+      }
+
+      if (scrollTop + clientHeight < offsetTop + offsetHeight) {
+        listboxElement.scrollTop = offsetTop + offsetHeight - clientHeight;
+      }
```

Comment:
This will make it able to scroll with keyboard
![autocomplete_keyboard](https://user-images.githubusercontent.com/40295569/194901261-773c319f-14d4-4ea6-9bbf-974d0cec04db.gif)